src/Network/NWEndpoint.cs

```diff
@@ -0,0 +1,163 @@
+
+		[Mac(10,14)][iOS(12,0)][Watch(5,0)][TV(12,0)]
+		[DllImport (Constants.NetworkLibrary)]
+		extern static NWEndpointType nw_endpoint_get_type (OS_nw_endpoint handle);
+
+		[Mac(10,14)][iOS(12,0)][Watch(5,0)][TV(12,0)]
+		public NWEndpointType Type => nw_endpoint_get_type (handle);
+
+		[Mac(10,14)][iOS(12,0)][Watch(5,0)][TV(12,0)]
+		[DllImport (Constants.NetworkLibrary)]
+		extern static OS_nw_endpoint nw_endpoint_create_host (string hostname, string port);
+
+		[Mac(10,14)][iOS(12,0)][Watch(5,0)][TV(12,0)]
+		public NWEndpoint Create (string hostname, string port)
+		{
+			if (hostname == null)
+				throw new ArgumentNullException (nameof (hostname));
+			if (port == null)
+				throw new ArgumentNullException (nameof (port));
+			var handle = nw_endpoint_create_host (hostname, port);
+			if (handle == IntPtr.Zero)
+				return null;
+			return new NWEndpoint (handle, owns: true);
+		}
+
+		[Watch (5,0), TV (12,0), Mac (10,14), iOS (12,0)]
+		[DllImport (Constants.NetworkLibrary)]
+		static extern string nw_endpoint_copy_port_string (OS_nw_endpoint endpoint);
+
+		[Watch (5,0), TV (12,0), Mac (10,14), iOS (12,0)]
+		public string Port => nw_endpoint_copy_port_string (handle);
+
+		
+		[Watch (5,0), TV (12,0), Mac (10,14), iOS (12,0)]
+		[DllImport (Constants.NetworkLibrary)]
+		static extern ushort nw_endpoint_get_port (OS_nw_endpoint endpoint);
+
+		[Watch (5,0), TV (12,0), Mac (10,14), iOS (12,0)]
+		public ushort PortNumber => nw_endpoint_get_port (handle);
+	
+		[Watch (5,0), TV (12,0), Mac (10,14), iOS (12,0)]
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_endpoint nw_endpoint_create_address (IntPtr sockaddrPtr);
+
+		// TODO: .NET has a SocketAddress type, we could use it for simplicity, but the
+		// address family would have to be mapped, and it does not look like a very useful
+		// type to begin with.
+
+		// extern char * _Nonnull nw_endpoint_copy_address_string (nw_endpoint_t _Nonnull endpoint) __attribute__((availability(tvos, introduced=12.0))) __attribute__((availability(watchos, introduced=5.0))) __attribute__((availability(ios, introduced=12.0))) __attribute__((availability(macos, introduced=10.14)));
+		[Watch (5,0), TV (12,0), Mac (10,14), iOS (12,0)]
+		[DllImport (Constants.NetworkLibrary)]
+		static extern string nw_endpoint_copy_address_string (OS_nw_endpoint endpoint);
+		
+		[Watch (5,0), TV (12,0), Mac (10,14), iOS (12,0)]
+		public string Address => nw_endpoint_copy_address_string (handle);
+		
+		// extern const struct sockaddr * _Nonnull nw_endpoint_get_address (nw_endpoint_t _Nonnull endpoint) __attribute__((availability(tvos, introduced=12.0))) __attribute__((availability(watchos, introduced=5.0))) __attribute__((availability(ios, introduced=12.0))) __attribute__((availability(macos, introduced=10.14)));
+		[Watch (5,0), TV (12,0), Mac (10,14), iOS (12,0)]
+		[DllImport (Constants.NetworkLibrary)]
+		static extern IntPtr /* struct sockaddr* */ nw_endpoint_get_address (OS_nw_endpoint endpoint);
```

Comment:
This isn't used/exposed, in which case it should probably be commented out.

Reply:
That is a work in progress